nominal/core/data_review.py

```diff
@@ -0,0 +1,134 @@
+            priority=checklist._conjure_priority_to_priority(check_alert.priority)
+            if check_alert.priority is not scout_checks_api.Priority.UNKNOWN
+            else None,
+        )
+
+
+@dataclass(frozen=True)
+class DataReviewBatchBuilder:
+    integration_rids: list[str]
+    _requests: list[scout_datareview_api.CreateDataReviewRequest]
+    _clients: DataReview._Clients = field(repr=False)
+
+    def add_integration(self, integration_rid: str) -> DataReviewBatchBuilder:
+        self.integration_rids.append(integration_rid)
+        return self
+
+    def add_request(self, run_rid: str, checklist_rid: str, commit: str) -> DataReviewBatchBuilder:
+        self._requests.append(scout_datareview_api.CreateDataReviewRequest(checklist_rid, run_rid, commit))
+        return self
+
+    def initiate(
+        self, wait_for_completion: bool = True, wait_timeout: timedelta = timedelta(minutes=1)
+    ) -> Sequence[DataReview]:
+        """Initiates a batch data review process.
+
+        Args:
+            wait_for_completion (bool): If True, waits for the data review process to complete before returning.
+                                        Default is True.
+            wait_timeout (timedelta): The maximum time to wait for the data review process to complete.
+                                      Default is 1 minute.
+
+        Raises:
+            TimeoutError: If the data review process does not complete before the wait_timeout.
+        """
+        request = scout_datareview_api.BatchInitiateDataReviewRequest(
+            notification_configurations=[
+                scout_integrations_api.NotificationConfiguration(c) for c in self.integration_rids
+            ],
+            requests=self._requests,
+        )
+        response = self._clients.datareview.batch_initiate(self._clients.auth_header, request)
+
+        if not wait_for_completion:
+            return [
+                DataReview._from_conjure(self._clients, self._clients.datareview.get(self._clients.auth_header, rid))
+                for rid in response.rids
+            ]
+
+        started = datetime.now()
+        completed_review_rids = []
+        completed_reviews = []
+        while datetime.now() - started <= timedelta(seconds=wait_timeout.total_seconds()):
+            sleep(2)
+            for rid in response.rids:
+                if rid not in completed_review_rids:
+                    review_response = self._clients.datareview.get(self._clients.auth_header, rid)
+                    review = DataReview._from_conjure(self._clients, review_response)
+                    if review.completed:
+                        completed_review_rids.append(rid)
+                        completed_reviews.append(review)
+            if len(completed_reviews) == len(response.rids):
+                return completed_reviews
+
+        raise TimeoutError(f"Data review initiation did not complete before wait_timeout. Review rids: {response.rids}")
```

Comment:
The built-in TimeoutError is for system functions, we shouldn't re-use that for our own timeout error: https://docs.python.org/3/library/exceptions.html#TimeoutError